lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -417,6 +417,9 @@ class TypeAnalyzer : public ForwardDataFlowAnalysis<ValueKnowledge> {
       return visitAtenBmmOp(bmm, operands);
     } else if (auto matmul = dyn_cast<AtenMatmulOp>(op)) {
       return visitAtenMatmulOp(matmul, operands);
+    } else if (auto mean = dyn_cast<AtenMeanOp>(op)) {
+      Type dtype = operands[0]->getValue().dtype;
```

Comment:
I just realized that the reduction ops had been ignoring the `dtype` argument (https://github.com/llvm/torch-mlir/pull/449). You should do something like 

https://github.com/llvm/torch-mlir/blob/c965f9f90f544de14fa0ccf216ace3616891da79/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L317-L321

here.

Reply:
(after https://github.com/llvm/torch-mlir/pull/449 lands)